design-proposals/migration-target.md

```diff
@@ -0,0 +1,216 @@
+- Workload balancing solution doesn't always work as expected
+  > I have configured my cluster with the descheduler and a load aware scheduler (trimaran), thus by default, my VMs will be regularly descheduled if utilization is not balanced, and trimaran will ensure that my VMs will be scheduled to underutilized nodes. Often this is working, however, in exceptional cases, i.e. if the load changes too quickly, or only 1 VM is suffering, and I want to avoid that all Vms on the cluster are moved, I need - for exception - a tool to move one VM, once to deal with this exceptional situation.
+- Troubleshooting a node
+- Validating a new node migrating there a specific VM
+  > I, as a battle tested admin, ordered a new node, because the old one broken, got the HW team to install it in the dc, ask the net team to wire it up, installed it myself, and included it in the cluster - tainted.
+  I drink a coffee, as this was a lot of work.
+  Now I want to move one of my own VMs over to it to do some sanity testing.
+  No other Vm on my 50 node cluster should be impacted. I just want my very own vm to move over and see if all behaves well.
+  I've seen enough issues with misbehaving hardware, misbehaving live migrations, cpu firmware issues, broken storage etc. this is my sanity test
+
+> [!IMPORTANT]
+> Directly selecting named nodes as destinations is not assumed to be a default tool for balancing workloads or all the use-cases above. It's instead just a convenient tool for exceptional situations and one-offs to ensure that and admin can quickly react to emergencies, and spikes.
+This proposal is part of a larger scheduling enhancement picture.
+Cluster admins are also looking for, for instance [descheduler integration](https://github.com/kubevirt/community/pull/2580), [load aware scheduling](https://github.com/kubevirt/user-guide/pull/621).
+This proposal is just an additional piece of the puzzle but, again, is the tool for exceptions and the corner cases, not the norm.
+
+Such a capability is expected from traditional virtualization solutions but, with certain limitations, is still pretty common across the most popular cloud providers (at least when using dedicated and not shared nodes).
+- For instance on Amazon EC2 the user can already live-migrate a `Dedicated Instance` from a `Dedicated Host` to another `Dedicated Host` explicitly choosing it from the EC2 console, see: https://repost.aws/knowledge-center/migrate-dedicated-different-host
+- also on Google Cloud Platform Compute Engine the user can easily and directly live-migrate a VM from a `sole-tenancy` node to another one via CLI or REST API, see: https://cloud.google.com/compute/docs/nodes/manually-live-migrate#gcloud
```

Comment:
I think this is an entirely different offering from what KubeVirt in general does.
I don't know how relevant is this example. Here the user specifically knows what node was given to him.

`Sole-tenancy lets you have exclusive access to a sole-tenant node, which is a physical Compute Engine server that is dedicated to hosting only your project's VMs`

Reply:
Why do you think that this use case is different from `Dedicated Hosts` on AWS EC2?
See for instance figure 1 on https://cloud.google.com/compute/docs/nodes/sole-tenant-nodes
Normally your VMs are going to be executed on a multi-tenant host shared with other customers.
If you have special requirements in terms of physical isolation you can decide to pay more and have a set of physical hosts that are exclusively dedicated to you, in that case you know the names of the hosts and you can also decide (for various reasons) to manually live migrate a VM to a named node within your sole-tenancy node group.

Now if you are the cluster admin of an on-premise (or not but with dedicated hosts) cluster with KubeVirt, as for this proposal, you will be able to live migrate a VM to a named node.
The assumption is still that you are the cluster admin and you know the name of the hosts in your cluster.
Why is it different?